ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/AttestationUtil.java

```diff
@@ -97,12 +104,18 @@ public static boolean verifyAttestation(BeaconState state, Attestation attestati
 
   // Get attestation data that does not include attester specific shard or crosslink information
   public static AttestationData getGenericAttestationData(BeaconState state, BeaconBlock block) {
-    // Get variables necessary that can be shared among Attestations of all validators
     UnsignedLong slot = state.getSlot();
+    return getGenericAttestationData(state, block, slot);
+  }
+
+  // Get attestation data that does not include attester specific shard or crosslink information
+  public static AttestationData getGenericAttestationData(
+      BeaconState state, BeaconBlock block, UnsignedLong slot) {
+    // Get variables necessary that can be shared among Attestations of all validators
     Bytes32 beacon_block_root = block.signing_root("signature");
     UnsignedLong start_slot = compute_start_slot_at_epoch(get_current_epoch(state));
     Bytes32 epoch_boundary_block_root =
-        start_slot.compareTo(slot) == 0
+        start_slot.compareTo(slot) <= 0
```

Comment:
I think, looking at the spec, this should have stayed as ==. no?

Reply:
Discussed offline - I'll add a ticket to rework this so that we use `==` here.  

The issue is that the test utility `AttestationGenerator` will try to generate attestations for any slot in the current epoch.  So, it can end up generating attestations for a future slot that doesn't correspond to the current block / state.  But a better way of handling this would be to just process the block / state in the test code so that we don't need to modify this logic.

Reply:
Ticket: https://pegasys1.atlassian.net/browse/BC-114